src/Microsoft.AspNet.Razor.Runtime/TagHelpers/TargetElementAttribute.cs

```diff
@@ -0,0 +1,44 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.AspNet.Razor.TagHelpers;
+
+namespace Microsoft.AspNet.Razor.Runtime.TagHelpers
+{
+    /// <summary>
+    /// Provides an <see cref="ITagHelper"/>'s target.
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Class, AllowMultiple = true, Inherited = false)]
+    public sealed class TargetElementAttribute : Attribute
+    {
+        /// <summary>
+        /// Instantiates a new instance of the <see cref="TargetElementAttribute"/> class.
+        /// </summary>
+        public TargetElementAttribute()
+        {
+            Tags = TagHelperDescriptorProvider.CatchAllDescriptorTarget;
+        }
+
+        /// <summary>
+        /// Instantiates a new instance of the <see cref="TargetElementAttribute"/> class.
+        /// </summary>
+        /// <param name="tags">
+        /// A comma-separated <see cref="string"/> of tag names the <see cref="ITagHelper"/> targets.
+        /// </param>
+        public TargetElementAttribute(string tags)
+        {
+            Tags = tags;
+        }
+
+        /// <summary>
+        /// A comma-separated <see cref="string"/> of tag names the <see cref="ITagHelper"/> targets.
+        /// </summary>
+        public string Tags { get; set; }
+
+        /// <summary>
+        /// A comma-separated <see cref="string"/> of attribute names the <see cref="ITagHelper"/> targets.
+        /// </summary>
+        public string Attributes { get; set; }
```

Comment:
In order to abide by spec, will add an extra ctor for `tags, attributes` but keep them both as strings . Also will remove the setter on `Tags`.